tools/engine_tool/lib/src/commands/build_command.dart

```diff
@@ -0,0 +1,108 @@
+//    "mac_host_engine" both define "host_debug".
+//  - Naming: Should BuildConfig be BuilderConfig and GlobalBuild be BuildConfig?
+//  - The BuildConfig database seems to be lacking things I expected:
+//    - There is no "android_debug_*" GlobalBuild in any of the builders that run on mac?
+//    - The gn flags specified in the GlobalBuilds don't reference rbe or goma? Should this be a local configuration that the user selects?
+//    - Need to better understand available targets per GlobalBuild. The lists seem slim.
+
+// TODO(johnmccutchan): List all available build targets and allow the user
+// to specify which one(s) we should build on the cli.
+
+// TODO(johnmccutchan): Can we show a progress indicator like 'running gn...'?
+
+/// The root 'build' command.
+final class BuildCommand extends CommandBase {
+  /// Constructs the 'query' command.
+  BuildCommand({
+    required super.environment,
+    required Map<String, BuildConfig> configs,
+  }) {
+    builds = runnableBuilds(environment, configs);
+    // Add options here that are common to all queries.
+    argParser.addOption(
+      _configFlag,
+      abbr: 'c',
+      defaultsTo: 'host_debug',
+      help: 'Specify the build config to use',
+      allowed: <String>[
+        for (final GlobalBuild config in runnableBuilds(environment, configs))
+          config.name,
+      ],
+      allowedHelp: <String, String>{
+        for (final GlobalBuild config in runnableBuilds(environment, configs))
+          config.name: config.gn.join(' '),
+      },
+    );
+  }
+
+  // List of compatible builds.
+  late final List<GlobalBuild> builds;
+
+  @override
+  String get name => 'build';
+
+  @override
+  String get description => 'Builds the engine';
+
+  Future<int> _gn(GlobalBuild build) async {
+    final String gnPath =
+        path.join(environment.engine.flutterDir.path, 'tools', 'gn');
+    final List<String> commandLine = [gnPath, ...build.gn];
+    final ProcessRunnerResult r = await environment.processRunner
+        .runProcess(commandLine, workingDirectory: environment.engine.srcDir);
+    if (r.exitCode != 0) {
+      environment.stderr.write('Failed to execute: ${commandLine.join(' ')}\n'
+          'stdout: ${r.stdout}\n'
+          'stderr: ${r.stdout}\n'
+          'exitCode: ${r.exitCode}');
+      return r.exitCode;
+    }
+    return 0;
+  }
+
+  Future<int> _ninja(GlobalBuild build) async {
+    final List<String> commandLine = ['ninja', '-C', build.name];
+    final ProcessRunnerResult r = await environment.processRunner
+        .runProcess(commandLine, workingDirectory: environment.engine.outDir);
+    if (r.exitCode != 0) {
+      // TODO(johnmccutchan): We probably want to stream the ninja output
+      // so we can see build progress and compilation failures immediately.
+      environment.stderr.write('Failed to execute: ${commandLine.join(' ')}\n'
+          'stdout: ${r.stdout}\n'
+          'stderr: ${r.stdout}\n'
+          'exitCode: ${r.exitCode}');
```

Comment:
This also loses output colors, which are really helpful. You probably want something like:

```dart
Future<int> runProcess(String executable, List<String> arguments) async {
  var process = await Process.start(
    executable,
    arguments,
    runInShell: true,
    mode: ProcessStartMode.inheritStdio,
  );

  return await process.exitCode;
}
```